### DISCUSSION TITLE
A miner is mining empty blocks

### DISCUSSION THREAD

**Discussion by didlie** [dogecoin/dogecoin#2591] [Help needed]
The network is frequently exploited by miner: DMr3fEiVrPWFpoCWS958zNtqgnFb7QWn9D

The miner appears to exploit the network by starting the hashing process on receipt of unconfirmed blocks, by mining a sequential empty block and submitting the new empty block as soon as the received block in confirmed. The miner is able to steal from the network by mining multiple blocks using (what appears to be) competitive mining hardware to mine sequential blocks before the propagation the most recent block. The miner is not contributing to the network because it consistently mines blocks where the only transaction is the receipt of the 10,000 Dogecoin mining network tax. The miner is not processing other transactions, but is able to strategically exploit and tax the Dogecoin network through advance-mining of empty sequential blocks.

---Suggested network changes to inhibit advance-mining of empty blocks---

1) There should be a network delay of 10 minutes for the acceptance of empty, transaction-less, blocks by the Dogecoin Core network.

**Comment by vertiond** [comment]
This is called block withholding and is a technique that can be advantageous over time when done correctly.  I have confirmed F2Pool to do this practice on the Vertcoin network in the past.  This isn't a network exploit because this miner is simply making the choice to not relay a solved block and instead get started on the next block immediately without spending time doing relay.  Remember that it's a miner's choice to include txs in their blocks.  Ways to help mitigate this is more network hash rate in order to lessen the block withholder's ability to do this and setting fee policy to not be _too_ low such that tx inclusion is not worth it.

**Comment by patricklodder** [comment]
Some context:

1. [They don't just mine empty blocks](https://sochain.com/block/DOGE/355fc5d132dca47f47b2144a45df0698ff824e9e6287f33b0041f5467620d6ef)
2. This particular pool has a history of updating later than the others, so it's possible they are still on 1.14.2. That version has a problem that block relay slows down message processing, including incoming transactions. 
3. Pools will optimize to start work as soon as possible after finding a block (or someone else found one) to keep all that hardware productive; I cannot recall a pool that **waits** for relay to competitors before starting on the next block. 

Besides, even an empty block that does not cause a chaintip fork still helps securing the network.

So before calling anyone parasitic or accusing them of anything and even going as far as proposing consensus changes, without providing any evidence, please elaborate what you mean by the following, including the data you used to come to your conclusions:

1. "exploit the network by starting the hashing process on receipt of unconfirmed blocks"
2. "mining a sequential empty block and submitting the new empty block as soon as the received block [is] confirmed"
3. "steal from the network by mining multiple blocks"
4. "The miner is not contributing to the network"
5. "The miner is not processing other transactions"
6. "strategically exploit and tax the Dogecoin network through advance-mining"

**Comment by didlie** [comment]
The network will never mature because the parasitic blocks are contributing to excessive dilution.

**Comment by ReverseControl** [comment]
@patricklodder To @didlie's point, a significant number of their rewarded blocks appear to have mined no transactions beyond the coinbase. Yes, ``significant`` can be quantified precisely, but I have to write code to get the actual number and I am not up for it at the moment. I just took a look at which blocks have no fees added to the rewarded 10K DOGE. Its clearly noticeable if you do a visual inspection.

**Comment by didlie** [comment]
Empty block mining can and is done strategically. This particular miner mines almost entirely empty blocks. It is irrelevant who the miner is, only that they are imementing a technical startegy to exoit the network, tax the network, and create depreciation and dilution without contributing to the processing of transactions. Of course, mining an empty block will always be possible, but the network must take preference for actual transaction processing, that is the purpose of the mining and network itself. 

---Proposed network upgrade---
1) New blocks with the largest number of orders are prefered in descending order.
2) Empty blocks are delayed by 10 minutes, providing for the mining of transaction-providing blocks, and inhibiting the occasional double-block  intra-minute empty blocks that credit miners for exploiting the network.

**Comment by vertiond** [comment]
Why don't you start by answering Pat's questions?

**Comment by didlie** [comment]
Answering the questions:

1) It is possible this miner is also a member of one or more of the other pools, and found an exploit to send un-mined and/or unconfirmed blocks to process simultaneously through a private hardware pool.

2) The miner associated with this address mines almost entirely empty blocks. It may have mined transaction blocks in the past, but every block I have seen mined by this miner in the past 30 days has been empty, just a simple credit to the address of 10,000 (now giving it a $2000/ minute credit at current market prices, and well worth the effort to struggle to implement exploits.)

3) Every empty block mined while transactions are waiting to be processed delays the function of the network, credits the miner delaying transaction processing with 10,0000 Doge, and dilutes the value of Dogecoin. It is an exploitative theft, and is obviously parasitic.

4)The miner focuses on pushing empty blocks that credit itself, slow down transaction processing from other users by pushing it's empty blocks in place of real transactions, and causing extemporaneous dilution.

5) Look at its mined blocks. It found an exploit to process its own empty blocks. I've been watching it periodically for the past 30 days, and it is obviously the focus of this miner. Again, it's not an attack on the miner. This is a call to inhibit the implementation of this exploit.

6) multipool membership possible: when the difficulty is low on any empty block that block can be easily pushed through if a miner has advance notice of the block before confirmation. By focusing on mining empty blocks, a miner can take unconfirmed blocks from multiple pools and exploit when a sequential empty block can be solved faster than a block containing transactions can be solved.

**Comment by patricklodder** [comment]
Note: you failed to include ANY evidence, but I'll indulge your speculation for the sake of clearing this up. Maybe, if you anger me sufficiently, I will do the research myself out of frustration with your panicking. Know that you are currently keeping me from doing important work though, so I beg you to please bring more evidence, not just accuse and speculate.

> every block I have seen mined by this miner in the past 30 days has been empty

I just linked you a block from today that wasn't empty in my first comment.

> It is an exploitative theft

If it is according to consensus rules then it's not theft, at worst it's a bug. If it's a bug, we need to look into fixing it. What's the bug?

> Every empty block mined while transactions are waiting to be processed delays the function of the network

Correct. But that doesn't mean there is malicious intent. It can be a misconfiguration, problems because of 1.14.2 that were fixed in 1.14.3, or an optimization that the pool is of the opinion does not hurt.

> The miner focuses on pushing empty blocks that credit itself

Every block credits the miner, why would they send the rewards for mining a block to someone else? That's possible and sub-optimal, but not violating consensus rules. If they would do it too often, truly stalling transaction processing, they would cut their own profits because the price of DOGE would go down. So there is an economic disincentive to do that.

> It found an exploit to process its own empty blocks.

That's not an exploit. Look closely at [testnet](https://chain.so/testnet/doge).

> when the difficulty is low on any empty block that block can be easily pushed through if a miner has advance notice of the block before confirmation. By focusing on mining empty blocks, a miner can take unconfirmed blocks from multiple pools and exploit when a sequential empty block can be solved faster than a block containing transactions can be solved.

Correct. So do you expect this to be a bug in [DigiShield](https://github.com/dogecoin/dogecoin/blob/4c93783ab64ba96e60b7ea37202c072c66f3f544/src/dogecoin.cpp#L41-L87)?

> ---Proposed network upgrade---
>
> 1. New blocks with the largest number of orders are prefered in descending order.
> 2. Empty blocks are delayed by 10 minutes, providing for the mining of transaction-providing blocks, and inhibiting the occasional double-block intra-minute empty blocks that credit miners for exploiting the network.

1. I can exploit this by sending DOGE to myself and making full blocks, ie. this proposed rule incentivizes miners to spam the network
2. I can exploit this by sending DOGE to myself without making full blocks, this proposed rule is easily circumvented.

**Comment by patricklodder** [comment]
> @patricklodder To @didlie's point, a significant number of their rewarded blocks appear to have mined no transactions beyond the coinbase. Yes, `significant` can be quantified precisely, but I have to write code to get the actual number and I am not up for it at the moment. I just took a look at which blocks have no fees added to the rewarded 10K DOGE. Its clearly noticeable if you do a visual inspection.

Agree. I am not saying it's not happening. I do want to investigate it, but we have to break this cycle of accusations without evidence. It anyway doesn't help, because Dogecoin is permissionless, so if there is a bug, we identify it, fix it, and propose a soft fork, which will likely take a longer time. 

There is no need to get worked up and do the FUD thing. @langerhans has a gif for this.

**Comment by rnicoll** [comment]
My inclination here is they're badly configured. In particular, if we follow this back; Dogecoin blocks are typically mined by miners also mining Litecoin. I don't have the tools to hand to dump the Litecoin coinbases out of those blocks, but if I pull up recent Litecoin blocks ( https://chain.so/LTC ), it doesn't take long to find the same thing on Litecoin:

* https://chain.so/block/LTC/2128643
* https://chain.so/block/LTC/2128638
* https://chain.so/block/LTC/009b8e3bf4a14da017a9c408eba8f533af74c2b0e9a5acc0430785fc8798ac3c

I don't know who `LTCPodha2LAxSyBGsGfkWcmKDEB8cWrqLU` are, and frustratingly Google doesn't find them, but lets see if we can find out and go "Hey could you fix your relaying" first.

In terms of the proposal; blocks are strictly sequential, so delaying an empty block ten minutes actually means empty blocks are not accept until 10 minutes since the previous block, although distributed agreement on time is essentially impossible, so functionally that means we modify the difficulty target for empty blocks by 10x. That's a fairly drastic change, and if they are exploiting something, I presume they'd just pad each block with dummy transactions paying themselves until it past the threshold anyway. Inclusion of transactions is something we fix via economics not technology.

**Comment by patricklodder** [comment]
>  I don't have the tools to hand to dump the Litecoin coinbases out of those blocks

```
$ dogecoin-cli getblock b1f17e3e93cea21f4a3eb2d31aed4cfb7e3f76edbe94aa2fded920ae7c103f9b | jq -r .auxpow.tx.hex | xargs -n 1 litecoin-tx -json
{
    "txid": "835617cd23aee24b9fb6c2ee450e5dda74a1bab3a4ea470a6a27576e7afd0699",
    "hash": "835617cd23aee24b9fb6c2ee450e5dda74a1bab3a4ea470a6a27576e7afd0699",
    "version": 1,
    "size": 200,
    "vsize": 200,
    "weight": 800,
    "locktime": 0,
    "vin": [
        {
            "coinbase": "03bc7a2004614decdb2cfabe6d6d76f2bc62fd678529723dc342ce04f673ad914bd8bb4deb698c760af0c510717e4000000000000000042f4c502f0822f524ee01000000",
            "sequence": 4294967295
        }
    ],
    "vout": [
        {
            "value": 12.54015453,
            "n": 0,
            "scriptPubKey": {
                "asm": "OP_DUP OP_HASH160 57757ed9e2fa7d86636bdd8066078e22cfa19244 OP_EQUALVERIFY OP_CHECKSIG",
                "hex": "76a91457757ed9e2fa7d86636bdd8066078e22cfa1924488ac",
                "reqSigs": 1,
                "type": "pubkeyhash",
                "addresses": [
                    "LTCPodha2LAxSyBGsGfkWcmKDEB8cWrqLU"
                ]
            }
        },
        {
            "value": 0.00000000,
            "n": 1,
            "scriptPubKey": {
                "asm": "OP_RETURN aa21a9ed01d40bee686956b60a860a15dba0d4107afd639799e31e0ff1c90ac5346930e0",
                "hex": "6a24aa21a9ed01d40bee686956b60a860a15dba0d4107afd639799e31e0ff1c90ac5346930e0",
                "type": "nulldata"
            }
        }
    ],
    "hex": "01000000010000000000000000000000000000000000000000000000000000000000000000ffffffff4403bc7a2004614decdb2cfabe6d6d76f2bc62fd678529723dc342ce04f673ad914bd8bb4deb698c760af0c510717e4000000000000000042f4c502f0822f524ee01000000ffffffff02ddc1be4a000000001976a91457757ed9e2fa7d86636bdd8066078e22cfa1924488ac0000000000000000266a24aa21a9ed01d40bee686956b60a860a15dba0d4107afd639799e31e0ff1c90ac5346930e000000000"
}
```

**Comment by patricklodder** [comment]
Also updated https://blockshibe.net to display empty blocks as yellow in the stack-map on the right so that there is live data viz. Will do more later.

**Comment by didlie** [comment]
This is seems an obvious bug. I know the value and utility of the network will improve if this exploit is disallowed.

I endorse the "10x difficulty on empty blocks" proposal.

Donations for discovering this bug are welcome:
DBJXH3mV154TcYHW6AYkM2v5jx4uj2p9QG

I am not a large holder, but I am banking on the value of the Dogecoin network in the long-term.

**Comment by patricklodder** [comment]
> Donations for discovering this bug are welcome:

What's the bug?

**Comment by RosellaGirl87** [comment]
and Elon Musk will launch a DOGE-1 rocket in Q1 2022 What do you think?

**Comment by rnicoll** [comment]
We've identified the pool, going to reach out to them.

**Comment by rnicoll** [comment]
Also having looked at this again, I'm not _that_ convinced we're not just seeing a pool getting lucky with finding the next block, and transaction rate is low enough their mempool is empty when they started working on it.

**Comment by victorsk2019** [comment]
Hello everyone,

I'm still learning how it all works but to address empty block mining issue, would it help to add a checking for empty blocks function in miner.cpp:

`bool BlockAssembler::CheckEmpty()
{
    return (nBlockTx == 0);
}`

And, if it returns 'true', then do not issue 10,000 coins to miner?  I realize this will not prevent someone from filling the block with fake transactions (as mentioned above already), but using safeguard function and not rewarding miner with 10,000 coins for mining empty blocks would perhaps help with this particular issue?

**Comment by didlie** [comment]
@victorsk2019 The issue with this fix is obviously that in the case transaction volume temporarily slows, miners maintaining the network will not be credited for keeping the network running.

---suggested solutions---
Scale difficulty by number of transactions per block, or rather total transaction fees collected, where the collected fees are credited to the miner in place of the flat 10,000 dilution. 
1) Least difficult: So, if the total fees collected are equal or greater than 10,000, the block difficulty will be set at a minimum and the miner will collect all the fees to total 10,000 or more (if transactions fees exceed 10,000). And in this case the network experiences zero dilution.
2) Intermediate difficulty: The total transactions fees are less than 10,000 and the miner is credited the full 10,000, with the dilution being the the difference of 10,000 - transaction fees = dilution of Dogecoin per mined block.
3) Most difficult: There are no transactions in the miner's block. The network experiences a full 10,000 dilution, and the miner is credited 10,000 for mining an empty block.

If possible to implement, this scheme should benefit the network by reducing dilution, inhibit exploitation of the network by miners possibly focusing their efforts on mining empty blocks in advance of legitimate transactions, and at the same time ensure the sustainability of the network by keeping the incentive to mine during a minute empty of submitted transactions.

**Comment by didlie** [comment]
@patricklodder @rnicoll @ReverseControl @vertiond 
yet another empty block by the same miner just now: Block #3912481

**Comment by quackduck** [comment]
> Hello everyone,
> 
> I'm still learning how it all works but to address empty block mining issue, would it help to add a checking for empty blocks function in miner.cpp:
> 
> `bool BlockAssembler::CheckEmpty() { return (nBlockTx == 0); }`
> 
> And, if it returns 'true', then do not issue 10,000 coins to miner? I realize this will not prevent someone from filling the block with fake transactions (as mentioned above already), but using safeguard function and not rewarding miner with 10,000 coins for mining empty blocks would perhaps help with this particular issue?

I mean, they're already incentivized to include TXs by the TX fees.

**Reply by didlie** [reply]
This change should be implemented to see if it solves the hack.

**Comment by didlie** [comment]
Are you serious? TX fees are tiny! Are you serious? $2000 + or - $0.40 is not incentive.
RE:
> > Hello everyone,
> > I'm still learning how it all works but to address empty block mining issue, would it help to add a checking for empty blocks function in miner.cpp:
> > `bool BlockAssembler::CheckEmpty() { return (nBlockTx == 0); }`
> > And, if it returns 'true', then do not issue 10,000 coins to miner? I realize this will not prevent someone from filling the block with fake transactions (as mentioned above already), but using safeguard function and not rewarding miner with 10,000 coins for mining empty blocks would perhaps help with this particular issue?
> 
> I mean, they're already incentivized to include TXs by the TX fees.

**Comment by rnicoll** [comment]
Talked to the miner. What's happening is this:

1. Block n is mined.
2. Nodes remove all transactions in block n from the mempool.
3. The miner's node sends the block template for n+1 is sent to their miners. As the mempool is empty, the template is empty.
4. If it's mined before the next template is sent, it generates an empty block.

The template refresh is up to 55 seconds currently, which means if a block is mined in under 55 seconds (give or take) it can come out empty.

So; they're looking at reducing the refresh interval. However, keep in mind this only happens because we have partially full blocks, if the blocks start filling up the "overflow" transactions will be in the next block.

**Comment by didlie** [comment]
This miner continues to generate the network, maybe one hundred or more empty blocks every day. The mining share has climbed over 30% from this miner and most blocks are empty. If this miner is allowed to continue it will ultimately lead to a fork, or rather it must lead to a hard fork. There is no "talking" to the miner. If you are talking to him, then you are part of the scam. And this an obviously parasitic scam from this miner.

**Reply by patricklodder** [reply]
> The mining share has climbed over 30% from this miner and most blocks are empty.

Agree that 30% is not cool. They have the hash power? Did you know these L7 ASICs are only shipping to customers in November? Maybe something is there? Not like that didn't happen before 😉

>  If this miner is allowed to continue [..]

It was never even allowed to start. That's what permissionless means: there is no allow or disallow, just free. Very contra to what a post-covid world looks like, but perhaps even more reason to cherish it. 

> If you are talking to him, then you are part of the scam.

Maybe over there, but over here it is not. I guess whomever talked should not come over to your place. They're welcome to come over to mine though. Sorry if that ruins your party.

> And this an obviously parasitic scam from this miner.

Once again. Look at the hash power. Make the calculations. Where does it come from?

**Reply by Scerini** [reply]
So I just remembered litecoinpool org reached out to me when we first discover this, here was their response:

Hi,

We are currently discussing this with Ross Nicoll, but in short: this is
an intended consequence of the strategy we follow to assemble and
distribute mining jobs to miners. When it helps reducing wait times (and
therefore the rate of orphaned blocks), our servers send out block
templates without non-coinbase transactions. We apply the same strategy to
all chains, but because Dogecoin has a short block time, the effect is
more noticeable there. Over the past month, over 40% of the Dogecoin
blocks we mined included at least two transactions.
Currently we have no reason to believe that this strategy could harm the
Dogecoin network, but as I said we are discussing this with the Dogecoin
dev team.

Best regards,
pooler
 

Also should be noted they are running v 1.14.4

**Reply by MishaBoar** [reply]
Great info @Scerini - I think this kind of thing is important to share with the community and must be pointed out and discussed. 

So, from your e-mail, it seems they imply that 60% of their mined block are empty? @patricklodder / @rnicoll  / @ReverseControl, what percentage of our blocks are currently empty? I tried to look it up on blockshibe but I cannot find the data from there. I can maybe write a little tool on dogebeard.

The combination of the short block time, low number of transactions on the network, and their own in-house optimization, combined with what I suppose is access to great infrastructure (sitting on a great internet node + running their own mining rigs nearby), can of course lead to them having an advantage over other pools in mining empty nodes. I mean, pools compete against each other to make sure miners want to mine in their pool over others. But if they are gobbling up significantly more empty blocks than others, then it is a problem. But I will leave technical analysis to the experts in here.

The problem is not "just" "harming the Dogecoin network", but _potentially_ harming the Dogecoin economy. Like most chains other than BTC, we do not have any study of outflows from miners. So even though there is no foul play involved, investigating this kind of stuff is important, if anything because the pool involved is potentially accumulating a lot of Dogecoin, which even if just invested in their own infrastructure and not used to provide liquidity to bad actors/speculators/etc., might lead to them gaining further advantage over time and to Doge flowing in the same pools of investors/holders etc. So, important to investigate also to prevent further centralization.

**Reply by patricklodder** [reply]
>  what percentage of our blocks are currently empty?

~11.5% over the last 2000 blocks. Updated blockshibe for you.

> But if they are gobbling up significantly more empty blocks than others, then it is a problem.

A political problem, maybe, if you make it one. Not so much a consensus problem right now. But agree that it should be monitored.

Since they are open about what's going on, ask them to change it. If they need help, I'll personally be willing to redirect some time spent here to help out. And I'm sure I'm not alone. Things can be improved, always, and it may (eventually) be in their interest too: Less negative sentiment = less downward pressure on exchange rate = more income. Everyone wins.

**Reply by qlpqlp** [reply]
Hello to all,

Im trying to understand how blockchain works in detail.

I think this recent transaction where in the same block  i sent 5 Doges to my wallet for more code testing and i notice this below.

Can someone explain how it manage to create an empty wallet with 10k Doge and sent with zero fees to the same wallet address that you are all talking in here.

Thanks in advance :)

![Screenshot_20211019-211218_Chrome.jpg](https://user-images.githubusercontent.com/5968076/137984222-99b678b0-c78e-4abb-9dc2-a57ec9b29f43.jpg)

**Reply by MishaBoar** [reply]
> ~11.5% over the last 2000 blocks. Updated blockshibe for you.

Patrick thanks for the update, and so fast. It would be interesting to see their current numbers (since @Scerini's e-mail is from a while ago) against this, because, if 60% of their total hashrate are empty blocks, this would basically mean that whatever setup they have in place is able to process almost all empty blocks before anybody else.

> A political problem, maybe, if you make it one. 

For sure, you are right.

> Since they are open about what's going on, ask them to change it. If they need help, I'll personally be willing to redirect some time spent here to help out. And I'm sure I'm not alone. Things can be improved, always, and it may (eventually) be in their interest too: Less negative sentiment = less downward pressure on exchange rate = more income. Everyone wins.

I think this is a good point. And their openness is commendable. Also, the issue is still a bit confined to this Github. If it gets more widespread attention, individual miners contributing to the pool that have at least a bit of interest in the health of Dogecoin and in holding it might switch pools, which could affect this pool negatively. So both for this and the reason you mentioned, a fix to this situation would be a win for everybody involved. It is great that potentially you can help them directly for this.

**Reply by patricklodder** [reply]
> It would be interesting to see their current numbers (since @Scerini's e-mail is from a while ago) against this

Click the miner addresses on there, because I knew you we're going to ask this, so I had added these counters to each miner's detailed info view too with that update 😁

>  If it gets more widespread attention, individual miners contributing to the pool that have at least a bit of interest in the health of Dogecoin and in holding it might switch pools, which could affect this pool negatively.

**Opinion:** I'm not entirely convinced of the premise that because there is an optimization that is [almost like the dogecoin core hardcoded reset default](https://github.com/dogecoin/dogecoin/blob/4c93783ab64ba96e60b7ea37202c072c66f3f544/src/rpc/mining.cpp#L1051-L1053), people should leave a pool, especially if that pool is talking to a dogecoin core maintainer, together looking into the matter. These things are hard to tune and hard to test too, especially because they are said not be not specific to Dogecoin on this one, so I'd give it some time.

**Reply by patricklodder** [reply]
> I think this recent transaction where in the same block i sent 5 Doges to my wallet for more code testing and i notice this below.
>
> Can someone explain how it manage to create an empty wallet with 10k Doge and sent with zero fees to the same wallet address that you are all talking in here.

Simple: if you mine the block, you'd pay the fees to yourself - the fee for the coinbase transaction shows as zero because there is no input. Its output is funded by block subsidy (10k) + fees (338.x). Works the same on all the bitcoin-derived chains.

**Reply by magneticat** [reply]
> Click the miner addresses on there, because I knew you we're going to ask this, so I had added these counters to each miner's detailed info view too with that update 😁

Oh yes! Thank you. At first I could not see it due to some caching in my browser I think.
 
> **Opinion:** I'm not entirely convinced of the premise that because there is an optimization that is [almost like the dogecoin core hardcoded reset default](https://github.com/dogecoin/dogecoin/blob/4c93783ab64ba96e60b7ea37202c072c66f3f544/src/rpc/mining.cpp#L1051-L1053), people should leave a pool, especially if that pool is talking to a dogecoin core maintainer, together looking into the matter. These things are hard to tune and hard to test too, especially because they are said not be not specific to Dogecoin on this one, so I'd give it some time.

I agree with you. I mean that people using other more chaotic channels of "disinformation" might jump to conclusions too soon if this gets out of Github or gets distorted in the wrong way. I think the pool is doing the right thing right now by discussing and explaining their side of things.

(this is Misha from my office, sorry)

**Reply by MishaBoar** [reply]
Back home! Very good to have this discussion - Blockshibe (I love it also on mobile - it looks beautiful) helps with this. Since we know they are mining 80% of the empty blocks, and since we know directly from them this is a strategy they have in place to optimize the work of their miners and it is not caused by the fact they are running older node versions, and since individual miners _seem_ to favor this pool over others (even though I am not sure how much of that hashrate is in-house)  because it is probably reliable and maybe a bit more profitable, I think it is very good to have this discussion with them.

Of course Doge is permissionless, so other pools can of course implement the same strategy, but at the same time I do not have the knowledge to know how much access to certain infrastructure is fundamental for it to work this well.

**Reply by patricklodder** [reply]
This pool operates a custom work distribution backend, as they are veterans in mining software, so I'm sure that they can figure something out - but nevertheless my offer to help if needed stands. 

Besides that, I think DOGE's position/influence over auxpow miners (which is 99% of pools) must have become significantly improved over the past, because:

On October 19th, 2020, the total subsidy from mining LTC + DOGE was approx:
- `48.10 * 12.50 * 576` = $346,320 from LTC, and
- `0.002590 * 10,000 * 1440` = $37,296 from DOGE

Whereas on October 19th, 2021, it was approx:
- `188.94 * 12.5 * 576` = $1,360,368 from LTC, and
- `0.245788 * 10,000 * 1440` = $3,539,347.2 from DOGE

If the world makes sense - I know it doesn't sometimes, but let's stick to it for now - that shift in the share paid to miners should bring some negotiating power to the table, and shouldn't really have to be anything more than soft power.

**Reply by MishaBoar** [reply]
What a great point Patrick, thank you. We are writing them a (nice, as it should be) e-mail taking all of this into consideration.

**Reply by MishaBoar** [reply]
Hey @patricklodder / @rnicoll  we have a reply, stating that basically - right now - they do not see this, and delaying the occasional transaction for a minute - as an issue and that claims from @didlie are unsubstantiated, as are our other concerns, considering the low number of transactions on the network. Can we forward the e-mail to you so you can take a look, or do you want us to share it any other way? 

Coincidentally, their proportions of the empty blocks vs blocks with transactions has changed over the past 24 hours.

Thanks.

**Reply by patricklodder** [reply]
Can always look and form an opinion, feel free to forward. Who is "we"?

Note that they clearly stated in the email pasted above:

> When it helps reducing wait times (and therefore the rate of orphaned blocks), our servers send out block templates without non-coinbase transactions.

This should be taken into account! An orphaned block is MUCH worse than a block without transactions.

**Reply by MishaBoar** [reply]
Hey Patrick, got it! Exactly why we need professionals to look into the reply.

"We" as in @scerini, me, and the others that have helped contacting the miners in the past days; "normal" Dogecoin users. I do not think any of us is involved in mining, either. As you know, @scerini did a great job in contacting the pools for the node updates (he is Muffin on reddit)! We wrote the e-mail together, and @scerini sent it to them. Do you have an e-mail he can forward it to? Or reddit's messenger is fine as well?

**Reply by Scerini** [reply]
Just to dm via reddit patrick

**Reply by patricklodder** [reply]
I don't think there's anything in that email that is really wrong. What they say is basically what I just quoted - that they reduce orphan rate by doing the fastest-path calculation on a new block that comes in. Personally, I'd accept their explanation.

This is also in line with the few empty blocks that I have inspected since this issue was raised: all of them are quickly found - well within the minute mark - after another block. What this means is that you get the added security of the extra block on transactions that were recently mined - i.e. your transfer to an exchange will confirm sooner because of this, but then of course DigiShield will retarget and the next block will be harder. 

I think the reason for this triggering so often is because of DigiShield and the amplitude retargeting per block it does. With a static difficulty, like Litecoin and Bitcoin have, with periodic retargeting, this would in theory trigger it less. I think as a next step we must do an analysis of the realized block spacing, because perhaps this is something we'll have to address at a protocol level. I hope you understand that DigiShield was not mature when it was force-hard-forked into Dogecoin in 2014, and there may be optimizations that we can make there.

**Reply by MishaBoar** [reply]
Thanks for checking it and getting back to us. Yes, I do remember the rush to add DigiShield in back then, indeed.

This seems fine to me - we have both @patricklodder and @rnicoll confirming that this should not be a matter of concern, and I trust and respect their professional opinion. But if anybody else with enough technical knowledge wants to chime in and articulate a different opinion and base it on facts we might be missing, that will be welcome. In the end, we are here for the good of the community.

I have one question: why, in your opinion, other pools do not use a similar strategy? It is probably not our concern what strategy pools follow in a permissionless system unless it affects us negatively, of course - but I am asking, from a mining pool's strategy point of view, why are others not using the same strategy?

Thanks!

**Reply by patricklodder** [reply]
> confirming that this should not be a matter of concern

That's not what I said. What I said was basically that they're not bs-ing you. I think there **_is_** a problem and I think it's caused by DigiShield. I think it should be investigated. I think this is a weakness in Dogecoin, less in pool software. It's basically a cause and effect thing: is the cause something the pool software does, or is it something DigiShield does? I suspect it's the latter.

> Why, in your opinion, other pools do not use a similar strategy?

They probably don't have an as highly optimized, custom, work distribution backend.

**Reply by MishaBoar** [reply]
> That's not what I said. What I said was basically that they're not bs-ing you. I think there **_is_** a problem and I think it's caused by DigiShield. I think it should be investigated. I think this is a weakness in Dogecoin, less in pool software

Sorry I should have contextualized: I meant to say there is probably no need to be concerned about the intent of the pool (the claims from the original poster), not that there is not a problem that does need solving, either in their implementation or in Digishield.

So basically you recommend that the  Digishield implementation should be looked into, first?

**Reply by patricklodder** [reply]
Yes, I think making baseless claims, as they rightfully point out is being done in this thread, is not "the way". 

Instead, do the research: 

1. Check the block times intrinsically logged inside the blocks, check how these are spaced, compare receipt times on a couple of nodes in case the timestamps are off, check spacing over an extended period of time, check spacing of the empty blocks. 
2. Talk to DigiByte devs or check their source to see what they've done to improve, and most importantly, why they did that.
3. See if we even still need DigiShield - what are its benefits over a slower retargeting scheme for Dogecoin in 2021...

To start with. I'm sure there's many more things that can be figured out.

**Reply by rnicoll** [reply]
Can you quantify the loss you're asserting this is leading to? Because if you're arguing for a fork, I'd like to point out last hard fork https://digiconomist.net/mcxNOW-loses-350m-doge-announces-haircut/ happened, and it's not like that was the only loss, just the biggest headline. I hope the landscape is better than it was back then, but I do think you underestimate the risk & cost of a fork.

Also; so lets say the consensus changed so empty blocks are invalid. The miner changes their block template generation to add `n` (where `n` is the required number of transactions in a block for it to be valid) transactions from and to themselves to every block. They pay no fees on this because the fees go to the miner (themselves). All you've achieved is increasing spam load on the network.

Switching timing algorithm away from Digishield could change observed behaviour, but again cost vs benefit.

I think there's a point that's being missed that blocks do not _just_ exist to contain transactions, they also enforce the history up to that block, making it harder to perform a 51% attack. Empty blocks are not valueless work.

If we get to a point where there are transactions overflowing from the previous block, and a miner is intentionally ignoring them, it would be more of a concern. Bitcoin solves this by having expensive fees and less frequent blocks, but I don't think changing towards that is exactly better.

**Reply by patricklodder** [reply]
>  Because if you're arguing for a fork

I didn't say that. I said, do the research. No need to panic.

**Reply by rnicoll** [reply]
@patricklodder sorry meant OP, not you

**Reply by patricklodder** [reply]
Oh sorry. I was sure OP didn't mention DigiShield but I did, hence my confusion.

**Comment by didlie** [comment]
![Capture](https://user-images.githubusercontent.com/31107452/136712102-7e355154-0c61-44d0-ad67-8e87b3f5240a.PNG)
not acceptable !!

**Comment by kopeboy** [comment]
Loosing faith on Doge if some people can extract this much money disproportionaly

**Comment by guanhchan** [comment]
<img width="1152" alt="Screen Shot 2021-11-12 at 11 43 03 AM" src="https://user-images.githubusercontent.com/16890423/141525729-4dbd7507-8c10-4639-a1e9-2a6bda85d6b3.png"
<img width="1135" alt="Screen Shot 2021-11-12 at 11 38 26 AM" src="https://user
<img width="1124" alt="Screen Shot 2021-11-12 at 11 35 53 AM" src="https://user-images.githubusercontent.com/16890423/141525804-30d556a6-2bcd-41bd-bcd5-7a3d61372d85.png">
-images.githubusercontent.com/16890423/141525743-7765255e-3a92-40df-b2bc-d96b6a3394eb.png">
<img width="1115" alt="Screen Shot 2021-11-12 at 11 37 40 AM" src="https://user
<img width="1125" alt="Screen Shot 2021-11-12 at 11 36 10 AM" src="https://user-images.githubusercontent.com/16890423/141525791-4d041e35-3eb4-4927-9ce0-74fbd216c31a.png">

<img width="1125" alt="Screen Shot 2021-11-12 at 11 37 09 AM" src="https://user-
<img width="1122" alt="Screen Shot 2021-11-12 at 11 35 26 AM" src="https://user-images.githubusercontent.com/16890423/141525822-3aecc83a-b0dd-4780-86e4-639ef079155e.png">
images.githubusercontent.com/16890423/141525776-97a17267-ed7c-4859-9bf4-f9ffe2d9adf4.png">
-images.githubusercontent.com/16890423/141525762-15818871-5da6-4493-9ea1-ac170552513f.png">
>


Hello everyone,

I have been very interested in how the blockchain decided the winner or how the reward is given, this wallet address seem to be collecting(rewarded) on blocks that have low to no transactions. I don"t understand how this system choses the miners, I thought mining is a rigours tasks and requires a lot of hash power. Currently this address have been collecting about 30% of the block rewards, which seem to be a lot for a solo miner. Does this solo miner have some power algorithm to allow him to successfully mine all of these block? 
Sorry if this question has been answered, I did read the whole comment section and wasn't sure if I found the answer, so I decided look on the blockchain, and seem like it is still going on. If this is a bug, or error, this wallet is making bank. 
Best,
Guan

**Reply by patricklodder** [reply]
The "miner" is not a solo miner - it's a pool of many miners. Check out [their stats](https://www.litecoinpool.org/stats) to get more insight into the effects of pooling multiple miners' work.

**Reply by guanhchan** [reply]
Thank you for the reply, and the link is very helpful.
